source/HardwareSerial.cpp

```diff
@@ -31,276 +63,559 @@ HardwareSerial::~HardwareSerial()
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::Two;
+        break;
+    case SERIAL_5O1:
+        m_serialDevice->DataBits = 5;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::One;
+        break;
+    case SERIAL_6O1:
+        m_serialDevice->DataBits = 6;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::One;
+        break;
+    case SERIAL_7O1:
+        m_serialDevice->DataBits = 7;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::One;
+        break;
+    case SERIAL_8O1:
+        m_serialDevice->DataBits = 8;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::One;
+        break;
+    case SERIAL_5O2:
+        m_serialDevice->DataBits = 5;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::Two;
+        break;
+    case SERIAL_6O2:
+        m_serialDevice->DataBits = 6;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::Two;
+        break;
+    case SERIAL_7O2:
+        m_serialDevice->DataBits = 7;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::Two;
+        break;
+    case SERIAL_8O2:
+        m_serialDevice->DataBits = 8;
+        m_serialDevice->Parity = Windows::Devices::SerialCommunication::SerialParity::Odd;
+        m_serialDevice->StopBits = Windows::Devices::SerialCommunication::SerialStopBitCount::Two;
+        break;
+    default:
+        hr = E_INVALIDARG;
+    }
 
-        _storageUsed = true;
-        _storageIndex = 0;
+    return hr;
+}
 
-        // if nothing was read, return 0 and dont do anything
-        if (_storageCount == 0)
-        {
-            _storageUsed = false;
-            return -1;
-        }
+/// Free the serial device and associated resources on a best effort basis.
+void HardwareSerial::end(void)
+{
+    CloseUart();
+
+    g_pins.verifyPinFunction(m_txPinNo, FUNC_DIO, BoardPinsClass::UNLOCK_FUNCTION);
+    g_pins.verifyPinFunction(m_rxPinNo, FUNC_DIO, BoardPinsClass::UNLOCK_FUNCTION);
+}
+
+void HardwareSerial::flush(void)
+{
+    //concurrency::create_task(
```

Comment:
These are needed.   They are part of the logic that allows serial pins to be used as GPIO pins if serial is not in use.